File: BalanceBeam/BalanceBeamAMIB1/states.cpp
#include "states.h"

static const StateInfo state_infos[2] = {
  {IDLE::setup, IDLE::enter, IDLE::exit, IDLE::loop, IDLE::event},
  {BALANCEBEAM::setup, BALANCEBEAM::enter, BALANCEBEAM::exit, BALANCEBEAM::loop, BALANCEBEAM::event}
};

static const WireValue wire_values[0] = {
  
};

MasterManager<State, 2, 0> manager(0x30ce77aa, state_infos, wire_values, 0);

namespace IDLE {


void event(uint8_t ev) {
  switch (ev) {
  
  default:
    break;
  }
}


}
namespace BALANCEBEAM {


void event(uint8_t ev) {
  switch (ev) {
  case 0:
    events::homeRedCar();
    break;
  case 1:
    events::homeBlueCar();
    break;
  case 2:
    events::dropRed();
    break;
  case 3:
    events::dropBlue();
    break;
  case 4:
    events::testBlue();
    break;
  default:
    break;
  }
}


}

File: BalanceBeam/states.js

var IDLE = {
  id: 0,
  master: {
    values: {
      
    },
    events: {
      
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_IDLE = 0;
var BALANCEBEAM = {
  id: 1,
  master: {
    values: {
      
    },
    events: {
      homeRedCar: function homeRedCar() { manager.sendEvent(0, 1); },
      homeBlueCar: function homeBlueCar() { manager.sendEvent(1, 1); },
      dropRed: function dropRed() { manager.sendEvent(2, 1); },
      dropBlue: function dropBlue() { manager.sendEvent(3, 1); },
      testBlue: function testBlue() { manager.sendEvent(4, 1); }
    }
  },
  tablet: {
    values: {
      
    },
    events: {
      
    }
  }
};
var STATE_BALANCEBEAM = 1;

var STATES = {
  IDLE: IDLE,
  BALANCEBEAM: BALANCEBEAM
};
var manager = new Manager([IDLE, BALANCEBEAM]);
